pkg/ccl/changefeedccl/sink_cloudstorage.go

```diff
@@ -648,21 +662,55 @@ func (s *cloudStorageSink) flushFile(ctx context.Context, file *cloudStorageSink
 		return file.flushToStorage(ctx, s.es, dest, s.metrics)
 	}
 
-	s.flushGroup.Add(1)
-	go func() {
-		defer s.flushGroup.Done()
-		// NB: must use s.flushCtx; ctx may be short lived (i.e. cancelled).
-		if err := file.flushToStorage(s.flushCtx, s.es, dest, s.metrics); err != nil {
-			log.Errorf(ctx, "error flushing file to storage: %s", err)
-			// We must use the same type for error we store in flushErr.
-			s.flushErr.CompareAndSwap(nil, &flushError{error: err})
+	s.flushQueue.Lock()
+	defer s.flushQueue.Unlock()
+	if s.flushQueue.q == nil {
+		// Start a new go routine if the flush queue is empty.
+		s.flushGroup.Add(1)
+		go func() {
+			s.asyncFlush(file, dest)
+		}()
```

Comment:
Why is this not `go s.asyncFlush(file, dest)`?

Reply:
why not :)